### PR TITLE
process: adjust quote_cmd_arg for empty arguments

### PR DESCRIPTION
**Pull request by ryancole**
On Windows, `quote_cmd_arg` incorrectly handles "empty" command-line args, to the best of my knowledge.

If we use the scenario mentioned in joyent/node#7138, then it's true that the `lpCommandLine` param for `CreateProcessW` will be incorrectly formatted. Just to recap in this post, joyent/node#7138 says that if node calls `spawn(process.argv[0], ['foo', '', 'bar'])` then `quote_cmd_arg` will cause `lpCommandLine` for `CreateProcessW` to be `"node.exe foo  bar"`.

**Note the double spaces between `foo` and `bar`.**

In my tests, with a C# command-line argument, this extra space is stripped and the application only interprets 2 command-line args, even though we clearly want 3, from our node.js `spawn` call.

Now, in `quote_cmd_arg` the following check exists ...

```
  /*
   * Check if the string must be quoted;
   * if unnecessary, don't do it, it may only confuse older programs.
   */
  if (len == 0) {
    return target;
  }
```

I believe that this should be changed to actually insert an empty double quote, instead of skipped. Based on the comment, older programs may be confused, but C# programs are definitely broken. Also, whatever the OP in joyent/node#7138 was working with is broken - looks like another node app. It's completely valid on Windows to provide an empty command-line param, it just needs to be wrapped in quotes.

### CODE REVIEW

File: src/win/process.c [hunk 1]
```diff
@@ -430,12 +430,12 @@ WCHAR* quote_cmd_arg(const WCHAR *source, WCHAR *target) {
...
-   * Check if the string must be quoted;
-   * if unnecessary, don't do it, it may only confuse older programs.
-   */
   if (len == 0) {
-    return target;
+	  /* Need double quotation for empty argument */
+	  target[0] = L'"';
+	  target[1] = L'"';
+	  target += 2;
+	  return target;
```
Comment (saghul) [inline src/win/process.c]: Please use 2 spaces for indentation.

Comment (saghul) [inline src/win/process.c]: Also, how about this?

```
*(target++) = L'"';
*(target++) = L'"';
return target;
```

**Comment by ryancole** [comment]
@saghul I've fixed the spacing, I also adjusted the code to use the `*(target++) = L'"'` technique. As a side note, I've also tested these libuv changes from within node.js and the comand-line string looks good to me.

**Comment by saghul** [comment]
LGTM. @piscisaureus, can you please take a quick look?

**Comment by ryancole** [comment]
ping

**Comment by saghul** [comment]
Ok, lets land this. Can you squash your commits into one and rebase?

**Comment by ryancole** [comment]
Will do

Ryan

**Comment by ryancole** [comment]
@saghul alright, commits squashed and rebased.

**Comment by saghul** [comment]
Thanks Ryan, landed in https://github.com/joyent/libuv/commit/907b55e4bfe404377cbca0080b012c550a1be9ed.

**Pull request closed without merging**